diff --git a/Ue4Export/Exporter.cs b/Ue4Export/Exporter.cs
index 8a81bc7..6babe5d 100644
--- a/Ue4Export/Exporter.cs
+++ b/Ue4Export/Exporter.cs
@@ -499,7 +499,7 @@ namespace Ue4Export
 
 							mLogger?.Log(LogLevel.Information, $"  Saving texture {texture.Name}");
 
-							string outPath = Path.Combine(mOptions.OutputDirectory, $"{ConvertAssetPath(texture.GetPathName())}.png");
+							string outPath = Path.Combine(mOptions.OutputDirectory, Path.ChangeExtension(assetPath, ".png"));
 							success &= WriteTexture(bitmap, SKEncodedImageFormat.Png, outPath);
 						}
 
